Source/dom/models/Document.js

```diff
@@ -374,6 +386,26 @@ export class Document extends WrappedObject {
 
     msdocument.close()
   }
+
+  changeColorSpace(colorSpace, convert = false) {
```

Comment:
I'm wondering if it should be `convertColorSpace` and `convert` should just be `true` :thinking:

Is there a case where you'd use this instead of the `colorSpace` setter?

Reply:
I was thinking it could be awkward to have two different apis for assigning vs. converting. By having assigning and converting available via the same method you can switch the behaviour with a one liner:

```
doc.changeColorSpace(ColorSpace.P3, shouldConvert)
```

Rather than

```
if (shouldConvert) {
  doc.convertColorSpace(ColorSpace.P3)
} else {
  doc.colorSpace = ColorSpace.P3
}
```

🤔

Reply:
is there a use-case where `shouldConvert` would be dynamic? Tbh I'm still a little be confused why you'd not want to always convert?

Reply:
I chatted to the Mac devs a bit about this. It sounds like converting is a potentially destructive act when switching between managed color profiles (it walks the document and mutates all the color values), but assigning just sets the property. So in a sense the sort of switch you want to do, assigning vs converting, may depend on the current color profile value ... hence my thinking of adding the flag to the method call.

Basically reproducing some of the similar flow available by the UI:

![image](https://user-images.githubusercontent.com/1078571/59119822-b1336a00-894b-11e9-867e-6f805e1437ef.png)

Beyond that, from the point of view of a designer using Sketch, the pros and cons of assigning vs converting are currently beyond my understanding ...

Reply:
Ah, maybe ¯\_(ツ)_/¯. Let's keep it that way then

Reply:
I suppose to be super explicit we could have an api like,

```
document.colorProfile // getter
document.colorProfile = 'P3' /// Assigning setter
document.assignColorProfile('P3')
document.convertToColorProfile('P3')
```

🤔